includes/class-wcpdf-admin.php

```diff
@@ -672,6 +688,46 @@ public function delete_document() {
 		}
 	}
 
+	public function regenerate_document() {
+		if ( check_ajax_referer( "wpo_wcpdf_regenerate_document", 'security', false ) === false ) {
+			wp_send_json_error( array(
+				'message' => 'nonce expired',
+			) );
+		}
+		if ( empty($_POST['order_id']) || empty($_POST['document']) ) {
+			wp_send_json_error( array(
+				'message' => 'incomplete request',
+			) );
+		}
+		if ( !current_user_can('manage_woocommerce') ) {
+			wp_send_json_error( array(
+				'message' => 'no permissions',
+			) );
+		}
+
+		$order_id = absint($_POST['order_id']);
+		$document = sanitize_text_field($_POST['document']);
+
+		try {
+			$document = wcpdf_get_document( $document, wc_get_order( $order_id ) );
+			if ( !empty($document) && $document->exists() ) {
+				$document->regenerate();
+				$response = array(
+					'message' => $document->get_type()." regenerated",
+				);
+				wp_send_json_success($response);
+			} else {
+				wp_send_json_error( array(
+					'message' => 'document does not exist',
```

Comment:
this also happens when you click the delete button and then the regenerate button, but the above error is just internal, without feedback. I would suggest to remove (or hide) the regenerate button altogether when deleting a document

Reply:
Solved with: https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/commit/1196cb364f728f037e2d66ab25cb29352818cbe8